### PR TITLE
Fix Table aria attributes

### PR DESCRIPTION
**Pull request by jsomsanith**
**What is the problem this PR is trying to solve?**
#1199 explain accessibility issues. It refers to http://w3c.github.io/aria-practices/#grid.
This PR try to solve the accessibility on Table html.

Current markup is
```
<div role="grid" />
	<div role="row">
		<div role="columnheader"/>
		<div role="columnheader"/>
	</div>
	<div role="rowgroup">
		<div role="row">
			<div role="gridcell" />
			<div role="gridcell" />
		</div>
	</div>
</div>
```

**What is the chosen solution to this problem?**
Follow https://www.w3.org/TR/2018/NOTE-wai-aria-practices-1.1-20180726/examples/grid/dataGrids.html

New markup is
```
<div role="grid" aria-label="from props" aria-colcount="2" aria-rowcount="1" /> // label and col/row count
	<div role="rowheader"> . // row --> rowheader
		<div role="columnheader" aria-sort="none"/> // sortable but not current
		<div role="columnheader"/>
	</div>
	<div role="rowgroup">
		<div role="row" aria-rowindex="1" > // aria-rowindex
			<div role="gridcell" aria-colindex="1" /> // aria-colindex
			<div role="gridcell" aria-colindex="2" />
		</div>
	</div>
</div>
```

**Before submitting a pull request,** please complete the following checklist:

- [x] The existing test suites (`npm test`) all pass
- [x] For any new features or bug fixes, both positive and negative test cases have been added
- [ ] For any new features, documentation has been added
- [ ] For any documentation changes, the text has been proofread and is clear to both experienced users and beginners.
- [x] Format your code with [prettier](https://github.com/prettier/prettier) (`npm run prettier`).
- [x] Run the [Flow](https://flowtype.org/) typechecks (`npm run typecheck`).

edit:

closes: #1199

### CODE REVIEW

File: source/Table/defaultHeaderRowRenderer.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ export default function defaultHeaderRowRenderer({
   style,
 }: HeaderRowRendererParams) {
   return (
-    <div className={className} role="row" style={style}>
+    <div className={className} role="rowheader" style={style}>
```
Comment (backwardok) [inline source/Table/defaultHeaderRowRenderer.js]: `role="row"` was correct here. [`rowheader`](https://www.w3.org/TR/wai-aria/#rowheader) is for the cell that represents the heading for a row, if it exists.

Comment (jsomsanith) [reply source/Table/defaultHeaderRowRenderer.js]: Oh my bad, bad understanding of the value. Thanks

File: source/Table/Table.js [hunk 2]
```diff
@@ -378,19 +378,26 @@ export default class Table extends React.PureComponent {
       };
     });
 
+    const columns = this._getHeaderColumns();
```
[the remark this answers is not in the document]
Comment (jsomsanith) [reply source/Table/Table.js]: Good point, I'll just get the number of children

File: source/Table/Table.js [hunk 3]
```diff
@@ -378,19 +378,26 @@ export default class Table extends React.PureComponent {
...
     return (
       <div
+        aria-label={this.props['aria-label']}
+        aria-labelledby={this.props['aria-labelledby']}
+        aria-colcount={columns.length}
+        aria-rowcount={this.props.rowCount}
         className={cn('ReactVirtualized__Table', className)}
         id={id}
         role="grid"
+        tabIndex="0"
```
[the remark this answers is not in the document]
Comment (jsomsanith) [reply source/Table/Table.js]: 👍 I was testing something, adding it to the focus flow but i was a misundertanding of how SR works.

File: source/Table/Table.js [hunk 4]
```diff
@@ -378,19 +378,26 @@ export default class Table extends React.PureComponent {
...
 
+    const columns = this._getHeaderColumns();
+
     // Note that we specify :rowCount, :scrollbarWidth, :sortBy, and :sortDirection as properties on Grid even though these have nothing to do with Grid.
     // This is done because Grid is a pure component and won't update unless its properties or state has changed.
     // Any property that should trigger a re-render of Grid then is specified here to avoid a stale display.
     return (
       <div
+        aria-label={this.props['aria-label']}
+        aria-labelledby={this.props['aria-labelledby']}
```
[the remark this answers is not in the document]
Comment (jsomsanith) [reply source/Table/Table.js]: I add them in propTypes definition

**Comment by jsomsanith** [comment]
About the Grid component, it's something quite generic. I'm not sure about how we could do that. Adding an extra div to wrap the rows will make it not compatible with the Table implementation.

For now there is a workaround, passing your own cellRenderer (that's what Table does). Or use Table with disableHeader.

**Comment by koenoe** [comment]
This looks good to me and I could really use this. Can we merge this PR in?

**Comment by koenoe** [comment]
Thanks for merging it in! When can we expect a new release?

**Comment by wuweiweiwu** [comment]
@koenoe hopefully this weekend!

**Pull request merged**